Formula/jrnl.rb

```diff
@@ -16,79 +15,95 @@ class Jrnl < Formula
-    sha256 "e27001de32f627c22380a688bcc43ce83504a7bc5da472209b4c70f02829f0b8"
+    url "https://files.pythonhosted.org/packages/be/ed/5bbc91f03fa4c839c4c7360375da77f9659af5f7086b7a7bdda65771c8e0/python-dateutil-2.8.1.tar.gz"
+    sha256 "73ebfe9dbf22e832286dafa60473e4cd239f8592f699aa5adaf10050e6e1823c"
   end
 
   resource "pytz" do
-    url "https://files.pythonhosted.org/packages/ca/a9/62f96decb1e309d6300ebe7eee9acfd7bccaeedd693794437005b9067b44/pytz-2018.5.tar.gz"
-    sha256 "ffb9ef1de172603304d9d2819af6f5ece76f2e85ec10692a524dd876e72bf277"
+    url "https://files.pythonhosted.org/packages/82/c3/534ddba230bd4fbbd3b7a3d35f3341d014cca213f369a9940925e7e5f691/pytz-2019.3.tar.gz"
+    sha256 "b02c06db6cf09c12dd25137e563b31700d3b80fcc4ad23abb7a315f2789819be"
+  end
+
+  resource "pyxdg" do
+    url "https://files.pythonhosted.org/packages/47/6e/311d5f22e2b76381719b5d0c6e9dc39cd33999adae67db71d7279a6d70f4/pyxdg-0.26.tar.gz"
+    sha256 "fe2928d3f532ed32b39c32a482b54136fe766d19936afc96c8f00645f9da1a06"
+  end
+
+  resource "PyYAML" do
+    url "https://files.pythonhosted.org/packages/8d/c9/e5be955a117a1ac548cdd31e37e8fd7b02ce987f9655f5c7563c656d5dcb/PyYAML-5.2.tar.gz"
+    sha256 "c0ee8eca2c582d29c3c2ec6e2c4f703d1b7f1fb10bc72317355a746057e7346c"
   end
 
   resource "six" do
-    url "https://files.pythonhosted.org/packages/16/d8/bc6316cf98419719bd59c91742194c111b6f2e85abac88e496adefaf7afe/six-1.11.0.tar.gz"
-    sha256 "70e8a77beed4562e7f14fe23a786b54f6296e34344c23bc42f07b15018ff98e9"
+    url "https://files.pythonhosted.org/packages/21/9f/b251f7f8a76dec1d6651be194dfba8fb8d7781d10ab3987190de8391d08e/six-1.14.0.tar.gz"
+    sha256 "236bdbdce46e6e6a3d61a337c0f8b763ca1e8717c03b369e87a7ec7ce1319c0a"
   end
 
   resource "tzlocal" do
     url "https://files.pythonhosted.org/packages/cb/89/e3687d3ed99bc882793f82634e9824e62499fdfdc4b1ae39e211c5b05017/tzlocal-1.5.1.tar.gz"
     sha256 "4ebeb848845ac898da6519b9b31879cf13b6626f7184c496037b818e238f2c4e"
   end
 
+  resource "zipp" do
+    url "https://files.pythonhosted.org/packages/11/b5/89f3ab6d45b2709863761bab58c574b2344ef215749abb5407818c21c9ca/zipp-2.1.0.tar.gz"
+    sha256 "feae2f18633c32fc71f2de629bfb3bd3c9325cd4419642b1f1da42ee488d9b98"
+  end
+
   def install
     virtualenv_install_with_resources
   end
 
   test do
     (testpath/"write_journal.sh").write <<~EOS
       #!/usr/bin/expect -f
-      set timeout -1
-      spawn #{bin}/jrnl today: Wrote this fancy test.
-      expect -exact "Path to your journal file (leave blank for ~/journal.txt):"
-      send -- "#{testpath}/journal\n"
-      expect -exact "Enter password for journal (leave blank for no encryption): "
-      send -- "Homebrew\n"
-      expect "Do you want to store the password in your keychain?"
-      send -- "N\n"
-      expect -exact "Journal will be encrypted."
-      expect "Entry added to default journal"
+      set timeout 1s
+      spawn #{bin}/jrnl --version
+      expect -exact "jrnl version v2.2"
```

Comment:
Could we keep a test a bit more complex than simply checking the version?

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Good point! I'll leave that for @wren to pick up as I remember they mentioned that there was a reason they had changed the test.

Reply:
@elliotmassen I was having the same trouble as you were with the test. Any invocation of the app was hanging during my attempts. I didn't want the test to hold up fixing the formula (since `brew install jrnl` is currently definitely broken).

@Moisan @SMillerDev Is there any chance to get this merged with the current (admittedly lacking) test, with the intent to follow up with a better test soon after?

Reply:
Nope, experience tells us that "follow up with a better test" rarely happens and until it does we have no way to make sure it'll work.

Reply:
Would you accept a fix for the currently listed older version (v1.9.8) that someone else (not associated with our project) broke while we figure out the test situation for the newer version (v2.2)?

Reply:
Fixes are always welcome.